libs/interpret-vision/src/lib/VisionExplanationDashboard/Controls/FlyoutObjectDetection.tsx

```diff
@@ -100,6 +108,93 @@ export class FlyoutObjectDetection extends React.Component<
       });
     }
   }
+
+  private readonly callbackRef = (editorCallback: HTMLDivElement) => (this.setState({ editorCallback: editorCallback }));
+
+  public loadImageFromBase64(base64String: string, editor: Editor) { // onReady is a function/callable
+    const image = new Image();
+    image.addEventListener("load", (e) => {
+        editor.addContentSource(e.target as HTMLImageElement);
+        editor.AS.setSelectionMode(2);
+    });
+    image.src = `data:image/jpg;base64,${base64String}`;
+  }
+
+  public drawBoundingBoxes(item: IVisionListItem): void {
+
+    // Stops if the div container for the canvastools editor doesn't exist
+    if (!this.state.editorCallback) {
+      return;
+    }
+    // Removes any pre-existimg images due to previous render calls
+    this.state.editorCallback.innerHTML = "";
+
+    // Initializes CanvasTools-vott editor
+    var editor = new CanvasTools.Editor(this.state.editorCallback); // store as an instance variable
+    editor.enablePathRegions(true);
+
+    // Adds image to editor
+    this.loadImageFromBase64(item.image, editor);
+
+    // Initialize color constants
+    const Color = CanvasTools.Core.Colors.Color;
+    const theme = getTheme();
+
+    // Ensuring object detection labels are populated
+    if (!this.props.dataset.object_detection_predicted_y
+        || !this.props.dataset.object_detection_true_y
+        || !this.props.dataset.class_names) {
+      return;
+    }
+
+    // Retrieving labels for the image in the Flyout
+    const predictedY : number[][] = this.props.dataset.object_detection_predicted_y[item.index];
+    const trueY : number[][]  = this.props.dataset.object_detection_true_y[item.index];
+
+    // Draws bounding boxes for each predicted object
+    for (let oidx = 0; oidx < predictedY.length; oidx++) {
+
+      // Creating box region
+      let predObject = predictedY[oidx]
+      let predBox = new RegionData(predObject[1], predObject[2],
```

Comment:
Use the static "Build" methods of the RegionData class to create the region data (i.e. BuildRectRegionData). This will ensure the correct region type is set and create the correct position, size and point data from the arguments.
The current code is only creating a region with position/size data and not providing any point data or the region type.